File: backend/SNS.Application/Support/TicketMessages/Contracts/TicketMessageDto.cs
namespace SNS.Application.Support.TicketMessages.Contracts;

public sealed record TicketMessageDto(
    Guid Id,
    Guid TicketId,
    Guid SenderId,
    bool IsFromAgent,
    string MessageBody,
    DateTime SentAt,
    IReadOnlyList<TicketAttachmentDto> Attachments);

File: SNS.Web/src/app/support/ticket-messages/contracts/ticket-message.dto.ts
import { TicketAttachmentDto } from './ticket-attachment.dto';

export interface TicketMessageDto {
    id: string;
    ticketId: string;
    senderId: string;
    isFromAgent: boolean;
    messageBody: string;
    sentAt: string;
    attachments: TicketAttachmentDto[];
}
